Use shorter arrow functions where possible

Snuffleupagus opened the pull request:
For arrow functions that are both simple and short, we can avoid using explicit `return` to shorten them even further without hurting readability.

For the `gulp mozcentral` build-target this reduces the overall size of the output by just under 1 kilo-byte (which isn't a lot but still can't hurt).

Snuffleupagus committed:
Use shorter arrow functions where possible

For arrow functions that are both simple and short, we can avoid using explicit `return` to shorten them even further without hurting readability.

For the `gulp mozcentral` build-target this reduces the overall size of the output by just under 1 kilo-byte (which isn't a lot but still can't hurt).

Snuffleupagus commented:
/botio test

calixteman commented:
Could we use https://eslint.org/docs/latest/rules/arrow-body-style ?

timvandermeij reviewed (approved):
Looks good to me, with if possible the suggestion for the ESLint rule from Calixte applied since it'd be nice to enforce this automatically in linting. Thanks!

Snuffleupagus committed:
Remove the `no-confusing-arrow` ESLint rule since it's deprecated

Please see https://eslint.org/docs/latest/rules/no-confusing-arrow

Snuffleupagus committed:
Enable the `arrow-body-style` ESLint rule

This manually ignores some cases where the resulting auto-formatting would not, as far as I'm concerned, constitute a readability improvement or where we'd just end up with more overall indentation.

Please see https://eslint.org/docs/latest/rules/arrow-body-style

calixteman commented on src/core/evaluator.js:
```diff
@@ -1129,6 +1130,7 @@ class PartialEvaluator {
         case "Font":
           isSimpleGState = false;
 
+          // eslint-disable-next-line arrow-body-style
```
We could probably do something like
```js
promise = promise.then(() => this.handleSetFont....)
                               .then(loadedName = > { ... });
```

Snuffleupagus replied:
This is one of the cases where, as mentioned in the commit message, I don't feel that it's really a readability improvement to change that. (Since there's *multiple* Promises involved here.)

calixteman commented on src/core/evaluator.js:
```diff
@@ -1154,6 +1156,7 @@ class PartialEvaluator {
           if (value instanceof Dict) {
             isSimpleGState = false;
 
+            // eslint-disable-next-line arrow-body-style
```
Why not remove the return here ?

Snuffleupagus replied:
If I change this now I'll probably need to re-run all tests again, and I figure that we can "save" the bots a little bit here.
Hence I'll defer this for now, but perhaps it can be folded into a later patch (since I've noticed some other, smaller, clean-up opportunities in this file).

calixteman reviewed (approved):
LGTM. Thank you.
